zoe/covid-tracker-react-native
# v3 vaccine edit

# Description

Rejigs the add vaccine screen(s) as per the new design to gather further data.
 Full requirements here: 
https://www.notion.so/joinzoe/017c3de591604a7b89d0b69e3c37b279?v=0133357df3ed42e1bd42c1e0d7afcf45&p=8c3e4d43cdfe459da2d65f5426b1b94d
Design: https://www.figma.com/file/7TEcX7f6rw3Z0WYR4TpX3R/COVID-Symptom-Study-(App)---MASTER?node-id=6743%3A3918&viewport=-2281%2C354%2C0.5

## How to test locally
Run the corresponding BE branch if not merged https://github.com/zoe/covid-app/pull/387 . If merged, point against staging or use master.

Access page: home > profile > not had test.
Observe no vaccines + "Add Vaccine" button
Tap "Add Vaccine" button. US/GB users should have a blue link to another screen with image examples (SE not).
Add a DOSE ONE vaccine. To start, just add a date and save. This will let you see the incomplete vaccine/doses list, plus the prompt edit.
From here, edit vaccine and complete fields as required. Note changing the date in edit mode will prompt alert on save.
Delete vaccine now at bottom of dose in edit mode only.

## On what platform have you tested the change?

- [x] iOS - Emulator
- [ ] iOS - Physical device

## Screenshots

Vaccine with a dose that is missing a date (easiest way to replicate is to create a dose, save it, then manually delete its date in django admin)
![image](https://user-images.githubusercontent.com/3264113/105504026-31dce080-5cbf-11eb-899d-5cf77be8ece3.png)

Dose 1 complete
![image](https://user-images.githubusercontent.com/3264113/105504173-5c2e9e00-5cbf-11eb-83d3-9bdb5adf7664.png)

Both doses complete
![image](https://user-images.githubusercontent.com/3264113/105504244-72d4f500-5cbf-11eb-83c8-cc0d9d483201.png)

Example vaccine doc screen
![image](https://user-images.githubusercontent.com/3264113/105504704-fbec2c00-5cbf-11eb-8142-f6adbab1618e.png)

## Checklist

- [ ] I have updated mockServer
- [ ] I've added analytics as relevent
- [ ] I have run `npm test`
- [ ] I have run `npm run test:i18n`

## Out of scope and potential follow-up
Approach followed the Formik form model in which one parent form is passed down to the props, and reused props check if first dose and select properties based on that. Would be good to make this less rigid.
Setting second dose to 'no' on edit mode doesn't delete it yet. Ticket created.
